site.hs

```diff
@@ -21,88 +21,149 @@ main = hakyllWith config $ do
+            ctx <- projectsCtx . sortOn itemIdentifier <$> loadAll "projects/*.markdown"
+
+            makeItem ""
+                >>= loadAndApplyTemplate "templates/projects/list.html" ctx
+                >>= loadAndApplyTemplate "templates/boilerplate.html"   sponsors
                 >>= relativizeUrls
 
-    match "projects/*.markdown" $ compile pandocCompiler
+-- news ------------------------------------------------------------------------------------------------
+    match "news/**.markdown" $ compile pandocCompiler
+    categories <- buildCategories "news/**.markdown" (fromCapture "news/categories/**.html")
 
-    create ["projects/index.html"] $ do
+    tagsRules categories $ \category catId ->  compile $ do
+        news <- recentFirst =<< loadAll catId
+        let ctx =
+                listField "news" (newsWithCategoriesCtx categories) (pure news) <>
+                dateField "category" "%B %e, %Y"                                <>
+                defaultContext
+
+        makeItem ""
+            >>= loadAndApplyTemplate "templates/news/tile.html" ctx
+            >>= relativizeUrls
+
+    create ["news/index.html"] $ do
         route idRoute
         compile $ do
-            ctx <- projectsCtx . sortOn itemIdentifier <$> loadAll "projects/*.markdown"
-            sponsors <- sponsorsCtx ctx . sortOn itemIdentifier <$> loadAll "donations/sponsors/*.markdown"
+            sponsors <- sponsorsCtx . sortOn itemIdentifier <$> loadAll "donations/sponsors/*.markdown"
+            newsWithCategories <- recentFirst =<< loadAll "news/categories/**.html"
+            
+            let ctx =
+                    listField "categories" defaultContext (return newsWithCategories) <>
+                    defaultContext
 
             makeItem ""
-                >>= loadAndApplyTemplate "templates/projects/list.html"  sponsors
+                >>= loadAndApplyTemplate "templates/news/list.html"     ctx
                 >>= loadAndApplyTemplate "templates/boilerplate.html"   sponsors
                 >>= relativizeUrls
 
+-- templates -------------------------------------------------------------------------------------------
+    match "templates/*" $ compile templateBodyCompiler
     match "templates/**" $ compile templateBodyCompiler
 
+
+--------------------------------------------------------------------------------------------------------
+-- CONTEXT ---------------------------------------------------------------------------------------------
+--------------------------------------------------------------------------------------------------------
+
+-- sponsors --------------------------------------------------------------------------------------------
+-- | Partition sponsors into by level: monad, applicative, and functor
+-- Sponsors are listed in the footer template, which means we need this
+-- context for most pages. The first argument is another context so
```

Comment:
this comment will need to be updated

Reply:
Oo sure, hawk eyes 👀 thanks for catching it